Teacher lounge WIP

8bitgentleman opened the pull request:
This adds a whole new set of areas to the Greendale campus including several improvements for the campus that can be purchased with Borcher’s money from the greendale bursar.  Right now there are still several things  that need to be finished/resolved before this is ready.

didory123 commented:
Holy crap, this is incredible! I'll check this out when I have time later, but so far it looks like an impressive amount of work.

I know I keep saying I'm gonna jump back into hawkthorne, but I'm way too swamped with schoolwork right now and don't know when I'll be able to come back. Good news is, I'm properly learning how to do  some legit programming so hopefully when I get back (and I will) I won't be submitting some amateur spaghetti code.

8bitgentleman commented:
Thanks @didory123 !  The more tester on this the better, I'd be interested to hear your thoughts.  Looking forward to seeing you back around here.

CalebJohn commented:
This is so awesome, So much new content! Couple of small issues I had, 
* I had trouble exiting from the teachers lounge and the speak easy, I think the doors just need to be shifted a bit
* The any useful information for me speech option at the bursar doesn't have any text

I'm also thinking it be helpful for the player if we drop some hints about where to go next, for example after borchert gives the player some money it may be helpful if borchert mentions where the bursor can be found.

8bitgentleman commented:
@CalebJohn everything you mentioned should be fixed.

CalebJohn commented:
Great! I'll take a look at it tonight

CalebJohn commented on src/maps/admin-hallway.tmx:
```diff
@@ -105,5 +100,16 @@
   </object>
+  <object type="info" x="244" y="195" width="66" height="42">
+   <properties>
+    <property name="info" value="Do to a mix up in the records room all student records have been lost.  As you know, our student records are stored on a Microsoft Paint file, which the Dean was assured was future-proof. | Meanwhile, our extended warranty is not being honored by the restaurant that used to be Circuit City and we are asking every student to re-register for their classes."/>
```
This should say "Due to a mix up".

CalebJohn commented:
Besides the typo I made a comment on this all looks great! Great work @8bitgentleman lets get this merged in asap!

8bitgentleman commented:
Typo has been fixed.  Since this is such a large addition I'd really like to get as many eyes on this as possible.  If @didory123 @edisonout @niamu etc. could take a quick look and run-through of this hopefully we can minimize bugs/future bugfixes.

edisonout commented:
Haven't had a proper run through yet, but from a quick play through it looks awesome. I've already run into a few bugs (spelling, npcs running away while they're talking, getting stuck in an endless loop of Shirley's sandwiches dialogue and not being able to get away). 

There's so much of season five & six I don't remember/never actually watched so I'm sure some of the weird stuff I don't get is a reference to something.

Are we ever going to get round to fixing the quest module so we can have multiple quests/access quests from the command line? Because if so, I think we need to stop adding quests until that's done.

I have some work stuff that I've been putting off that needs doing by Monday, so I'll probably procrastinate this weekend and do some in-depth testing instead. 

I know I haven't been around for a while to help out or test things but in future I think we should probably do little changes and tests at a time rather than a huge new number of levels/quests because though it's an awesome new feature, it's likely to have lots of bugs we miss because there's so much going on.

8bitgentleman commented:
Yea this pull kind of got away from me.  I can definitely separate it into 3 pulls (1 for the lounge, 1 for the speakeasy and 1 for the bursar) if that would make things easier.  If I remember correctly quests from the command line aren't possible/ hideously difficult to implement.  I think it was @CalebJohn who I discussed this with.  Looking forward to seeing all the ways you can break this.

edisonout commented:
@8bitgentleman - don't worry about separating it out. We just need to think about keeping things small and separate in future.

I think I've tested everything for the teachers lounge section (except spelling, grammar and sound). I have just tried playing it rather than trying to break it.

#### Greendale exterior

- lots of no dean poster/skull posters - probably shouldn't be on the outside walls?
- hologram - does it ever do anything?
- Craig Cakowski faces the wrong way when you talk to him. If you talk about his flashlight, he'll start walking. Make his walking area smaller so he doesn't cover the doors & you end up going through them when you try to interact with him
- would it make sense to move the quad to the extra hole in the hedge by Luis Guzman? (not related to this pull request)

#### Teachers lounge

- would it be better if the room was a tile thinner so there wasn't the pan across & you could see the entire room in one go?
- door out of teacher's lounge is really wide, it's still hard to get in the bathroom
- You need to extend the info nodes down to the floor so you don't have to jump to read the back wall Split the text up rather than have it all show at once - looks very cramped. The bottom purple square doesn't have any text. 
- why isn't the teacher an NPC?
- what's the red thing in the top right hand corner?
- make the trigger come down to the floor for the vending machine

#### Teahers lounge bathroom

- Why does the player need to go in here? Any why need a flashlight? Is this something I'm forgetting?
- extend the lightswitch node down to the floor so people can interact with it
- how many keys do we have in the game? Is there going to be enough slots in the inventory? Why aren't we checking to see if a player has a material in their inventory instead?
- looks very incomplete. Is there anything to do other than get the rope? Could it at least be centred?
- the mirror & skirting are messed up and the duct stops in mid-air

#### Wings 1, 2 & 3

- why is the vent saying "hello, rich people?"
- they all looks like it's going to be 2.5D but is 2D
- any reason they're all one tile larger than screensize?
- looks like you should be able to pick up the bottles
- there's nothing to do in them
- the door trigger for the jukebox needs to come down to the floor - Annie isn't tall enough to activate it

#### Computer Lab
- again not 2.5D (although this looks like it would be complicated) & 1 tile wider than the window size so scrolls
- 4 lines of text in one go makes the dialogue really cramped
- that amount of money does not fit in the hud (is there a reason we get that amount of money? it seems absolutely ridiculous)
- you need some sort of message telling you you can't get out/why you can't get out & then the room should be small enough you can see the jukebox move when you get enough points

8bitgentleman commented:
Alright I've addressed some of the bug and concerns that you brought up:

- I'm a terrible speller, always have been.  Let me know what needs to be corrected.

#### Greendale exterior
- [x] removed skull posters, they were put there in mistake when the tileset changed slightly
- [x] You have to pay the bursar to get the hologram to work.  I've altered the text to inform the player that it's broken to make that more obvious
- [x] Craig's walking path has been adjusted
- [ ] I couldn't get Craig to face the wrong direction and walking while a prompt is open is an existing npc bug.
- I'm not a fan of moving the quad to behind the hedge.  You need to go there to complete quests and such so I don't think it should be hidden.

#### Teachers lounge
- [x] Fixed room size
- [x] Extended,split-up and added info nodes 
- [x] Removed red thing
- [x] Extended trigger
- [ ] The bathroom door is being quite difficult.  I'm not sure how to make entering it easier
- Professor Cornwallis isn't an NPC because I didn't have anything else for him to say.  I'm welcome to suggestions though.

#### Teachers lounge bathroom
- [ ] I'm confused as to how the mirror and skirting are messed up.  I'm using the same tileset as the other bathrooms.
- [ ] We do have a lot of keys in the game.  Doors are coded to look for keys not materials so that's why I went with that.  It could be changed it it's a big deal.
- [x] info node extended 
- The bathroom is part of a mini quest to allow the player to go down into the computer wing.  In the episode they needed rope and a flashlight ( If I remember correctly ) to go down there.  The player has to first find a flashlight ( from Craig outside ) to get into the bathroom and then use the rope in there to descend into the computer wing.
- I wanted the bathroom to be much smaller than the other ones since it's only for teachers.  I'm not sure how to center levels but that might make it seem less incomplete.

#### Wings, 1, 2, 3
- [x] vent fixed
- [x] room sizes fixed
- [x] tweaked trigger size
- The bottles are outside of the area that the player can access so I opted not to make them nodes.
- The rooms look like that because that was the art I was given.  Do you think it's confusing enough for it to be changed?
- The rooms are a bit empty simply because that's how they were in the episode.  Any ideas for stuff to add or should the 2nd room be axed?

#### Lab
- [x] Room size fixed
- [x] Overflowing text fixed
- [ ] I know there's a way to do it but I can't get the door to display text when it's closed
- Levels aren't 2.5D for the same reason as the wings
- The money is a bit of a joke, albeit a strange one.  1 million Gerald Ford are equivalent to $5,044,350 today with inflation.  Maybe that doesn't apply to actual cash?  I'm not sure.

Thanks for playing through all this, I'd be interested to see what bugs you find in the other new areas.

edisonout commented:
Thanks Gent. 

No worries about moving the quad. It was just an idea without much thought behind it. We should come up with something worth hiding in that third hedge hole though. I think NPCs facing the wrong direction is in the NPC code rather than anything you've done. I remember the alien quest guy being a pain with walking off while he was talking.

Regarding the rope and flashlight, I wasn't sure if that had anything to do with the episode or was just something extra for the players to do before they got into the computer wing. I really should watch the later episodes again (I've watched season 5 once and still haven't watched all of 6 - no clue what that giant hand statue is about). I understand the rooms being empty but it just seems a drag for the player to go lounge->exterior->lounge->bathroom->lounge->wing-1->wing-2->wing-3 and they still haven't accomplished anything or get rewarded until they get into the computer lab. Maybe I'm just not used to extended quests. It's not really a criticism, just something I'm not familiar with.

I hate making 2.5D rooms so I don't blame you. I just got confused when I tried to walk around and then ended up crouching instead.

didory123 commented:
@8bitgentleman Hey Gent, what else is left to do for this pull? If I could help out in any way, let me know

8bitgentleman commented:
Hey @didory123 I updated the top with open issues.  Also take a look at @edisonout and my comments.  There are some levels that could use secrets/rewards for getting to them so figuring those out would be helpful too.
